syntax = "proto3";

package product;

message Product {
    int32 id = 1;
    string name = 2;
    int32 quantity = 3;
}

message CreateProductRequest {
    Product product = 1;
}

message CreateProductResponse {
    Product product = 1;
}

message GetProductRequest {
    int32 id = 1;
}

message GetProductResponse {
    Product product = 1;
}

message ListProductsRequest {}

message ListProductsResponse {
    repeated Product products = 1;
}

message UpdateProductRequest {
    Product product = 1;
}

message UpdateProductResponse {
    bool success = 1;
}

service ProductService {
    rpc CreateProduct (CreateProductRequest) returns (CreateProductResponse);
    rpc GetProduct (GetProductRequest) returns (GetProductResponse);
    rpc ListProducts (ListProductsRequest) returns (ListProductsResponse);
    rpc UpdateProduct (UpdateProductRequest) returns (UpdateProductResponse);
}
